utils/ort/src/funTest/kotlin/storage/S3FileStorageFunTest.kt

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2023 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.utils.ort.storage
+
+import com.sun.net.httpserver.HttpExchange
+import com.sun.net.httpserver.HttpHandler
+import com.sun.net.httpserver.HttpServer
+
+import io.kotest.assertions.throwables.shouldNotThrowAny
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.Spec
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.core.test.TestCase
+import io.kotest.core.test.TestResult
+import io.kotest.matchers.shouldBe
+
+import java.net.HttpURLConnection
+import java.net.InetAddress
+import java.net.InetSocketAddress
+import java.util.Collections
+
+import kotlin.random.Random
+
+import org.apache.logging.log4j.kotlin.logger
+
+class S3FileStorageFunTest : WordSpec() {
+
+    private val loopback = InetAddress.getLoopbackAddress()
+    private val protocol = "http"
+    private val port = Random.nextInt(1024, 49152) // See https://en.wikipedia.org/wiki/Registered_port.
+        .also { logger.debug { "Using port $it for S3 Mock server." } }
+
+    private val bucket = "ort-scan-results"
+
+    private val handler = object : HttpHandler {
+        val requests = Collections.synchronizedMap(mutableMapOf<String, String>())
+
+        override fun handle(exchange: HttpExchange) {
+            when (exchange.requestMethod) {
+                "HEAD" -> {
+                    val status = if (requests.containsKey(exchange.requestURI.toString())) {
+                        HttpURLConnection.HTTP_OK
+                    } else {
+                        HttpURLConnection.HTTP_NOT_FOUND
+                    }
+                    exchange.sendResponseHeaders(status, -1)
+                }
+                "PUT" -> {
+                    val key = exchange.requestURI.toString().removePrefix("/$bucket/")
+                    requests[key] = exchange.requestBody.reader().use { it.readText() }.split("\n")[1].trimEnd()
+                    exchange.sendResponseHeaders(HttpURLConnection.HTTP_OK, 0)
+                }
+                "GET" -> {
+                    val key = exchange.requestURI.toString().removePrefix("/$bucket/")
+                    requests[key].let { data ->
```

Comment:
There doe snot seem to be a benefit by using `.let` here, so consider using plain `val data = requests[key]` to reduce nesting of following code.